python/bind_data_access.h

```diff
@@ -20,6 +20,31 @@ namespace py = pybind11;
 using namespace scipp;
 using namespace scipp::core;
 
+template <class Var> struct VarianceSetter {
+  template <class T> struct SetVariances {
+    static void apply(Var &var, py::array variances) {
+      py::array_t<T> variancesT(variances);
+      py::buffer_info info = variancesT.request();
+      auto dims = var.dims();
+      auto labels = dims.labels();
+      Vector<T> tmp(var.data().size());
```

Comment:
This line makes we wonder: Since we are making a copy from numpy anyway, why bother with all this code? Just do this below:

```cpp
 static void set_variances(Var &view, const py::array &data) {
    if (!view.hasVariances()) {
      view.setVariances(Vector<double>(view.dims().volume()); // or add method to default-init variances?
    }
    expect_shape_compatible(view, data);
    set(get<get_variances>(view), data);
}
```

And similar for `set_variance`.

Saves 30 lines of code and significantly reduces complexity?